app/src/main/java/com/example/habitsmasher/ListFragment.java

```diff
@@ -0,0 +1,90 @@
+package com.example.habitsmasher;
+
+import androidx.fragment.app.Fragment;
+import androidx.recyclerview.widget.LinearLayoutManager;
+import android.view.View;
+
+import com.google.firebase.firestore.FirebaseFirestore;
+import com.google.firebase.firestore.Query;
+
+/**
+ * Abstract class that specifies the behaviour of any fragment
+ * using a RecyclerView to display a list of objects of type T
+ * stored in a Firebase database. This displayed list supports
+ * adding, editing, deleting and viewing the details of items
+ * from the list
+ * @author Jason Kim
+ */
+public abstract class ListFragment<T> extends Fragment {
```

Comment:
The ListFragment abstract class is a generic class with type T, where is the type of object stored within the ListFragment. 
Thus, a fragment listing objects of type T would extend ListFragment<T>.